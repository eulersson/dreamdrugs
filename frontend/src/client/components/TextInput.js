import React from "react";
import PropTypes from "prop-types";

import "./TextInput.css";

function TextInput(props) {
  return (
    <input
      type="text"
      defaultValue={props.initial}
      onChange={e => props.onChange(e.target.value)}
    />
  );
}

TextInput.propTypes = {
  initial: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
}

export default TextInput;
