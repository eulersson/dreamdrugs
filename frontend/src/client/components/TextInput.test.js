import React from "react";
import TextInput from "./TextInput";

import renderer from "react-test-renderer";

import { shallow } from "enzyme";

describe("text input", () => {
  it("renders correctly", () => {
    const component = renderer
      .create(<TextInput initial="foo" onChange={(value) => {}} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it("default initial value actually works", () => {
    const component = shallow(<TextInput initial="potato" />);
    expect(component.find('input').first().prop("type")).toBe("text");
    expect(component.find('input').first().prop("defaultValue")).toBe("potato");
  });
});
