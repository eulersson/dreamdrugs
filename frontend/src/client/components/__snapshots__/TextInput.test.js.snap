// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text input renders correctly 1`] = `
<input
  defaultValue="foo"
  onChange={[Function]}
  type="text"
/>
`;
